aries_cloudagent/protocols/issue_credential/v2_0/formats/vc_di/handler.py

```diff
@@ -0,0 +1,577 @@
+
+        cred_def_id = await issuer.match_created_credential_definitions(
+            **cred_proposal_message.attachment(VCDICredFormatHandler.format)
+        )
+
+        async def _create():
+            offer_json = await issuer.create_credential_offer(cred_def_id)
+            return json.loads(offer_json)
+
+        multitenant_mgr = self.profile.inject_or(BaseMultitenantManager)
+        if multitenant_mgr:
+            ledger_exec_inst = IndyLedgerRequestsExecutor(self.profile)
+        else:
+            ledger_exec_inst = self.profile.inject(IndyLedgerRequestsExecutor)
+        ledger = (
+            await ledger_exec_inst.get_ledger_for_identifier(
+                cred_def_id,
+                txn_record_type=GET_CRED_DEF,
+            )
+        )[1]
+        async with ledger:
+            schema_id = await ledger.credential_definition_id2schema_id(cred_def_id)
+            schema = await ledger.get_schema(schema_id)
+        schema_attrs = set(schema["attrNames"])
+        preview_attrs = set(cred_proposal_message.credential_preview.attr_dict())
+        if preview_attrs != schema_attrs:
+            raise V20CredFormatError(
+                f"Preview attributes {preview_attrs} "
+                f"mismatch corresponding schema attributes {schema_attrs}"
+            )
+
+        cred_offer = None
+        cache_key = f"credential_offer::{cred_def_id}"
+
+        if cache:
+            async with cache.acquire(cache_key) as entry:
+                if entry.result:
+                    cred_offer = entry.result
+                else:
+                    cred_offer = await _create()
+                    await entry.set_result(cred_offer, 3600)
+        if not cred_offer:
+            cred_offer = await _create()
+
+        vcdi_cred_offer = {
+            "data_model_versions_supported": ["1.1"],
+            "binding_required": True,
+            "binding_method": {
+                "anoncreds_link_secret": {
+                    "cred_def_id": cred_offer["cred_def_id"],
+                    "key_correctness_proof": cred_offer["key_correctness_proof"],
+                    "nonce": cred_offer["nonce"],
+                },
+                "didcomm_signed_attachment": {
+                    "algs_supported": ["EdDSA"],
+                    "did_methods_supported": ["key"],
+                    "nonce": cred_offer["nonce"],
+                },
+            },
+            "credential": {
+                "@context": [
+                    "https://www.w3.org/2018/credentials/v1",
+                    "https://w3id.org/security/data-integrity/v2",
+                    {"@vocab": "https://www.w3.org/ns/credentials/issuer-dependent#"},
+                ],
+                "type": ["VerifiableCredential"],
+                "issuer": "public_did",
+                "credentialSubject": cred_proposal_message.credential_preview.attr_dict(),
+                "issuanceDate": "2024-01-10T04:44:29.563418Z",
```

Comment:
the issuance date must be generated of course

Reply:
Fixed, thank you!